### PR TITLE
Fix RC3 logic

### PR DESCRIPTION
**Pull request by klimofey**
- Add otional field "disable_sending_commands_when_device_off" to codes files
- Add possibility to use in temperature field values: string like common IR code and object
`{ 
"on": "dkawmdlkwamdlkawmdlawdlmaw", 
"regular": "wadmpp9iunjonj" 
}`
- Add 1948.json - RC3 for Moes UFO-R11

### CODE REVIEW

**Comment by litinoveweedle** [comment]
Hello,
thank you for PR. I would like to understand what is the use case you are proposing to implement. If I understand your changes correctly, you are proposing changes to implement very specific Electra IR remote control. It seems, that it differs a lot from how currently supported AC units works.

Could you please describe in detail how Electra IR works and what is the use case behind your PR?

P.S.: I am not too much in favor of implementing features used only on specific type of AC units. This would be nightmare to maintain in longer term. Also I am currently working on preset modes support and most of the changes are exactly at the same place like your PR.

**Comment by litinoveweedle** [comment]
Thank you for an explanation.
1. That is actually seems like bug. I will check this, I probably know from where this is coming from.
2. This is normal behavior and most of the AC unit are working in this way. Only one code including all the setting shall be sent.
3. Clear, thank you

**Comment by klimofey** [comment]
@litinoveweedle

1. ok
3. ok

________
2. Did not quite understand. Now if I send a command from the off state to cooling, for example, then the command cold + intensity + temperature is sent

But for the air conditioner to accept this command, it must first be turned on. If I add "on" next to the "off" command, it is sent every time the temperature changes. It turns off my air conditioner if it was on and vice versa.

That's why I added commands to turn it on from the off mode and switch the air conditioner already in the on mode. At the same time, I left the option of simply specifying the code in the temperature, and not the dictionary

**Comment by litinoveweedle** [comment]
So let me rephrase your issue. Your device has same command for "on" and "off", therefore if defined as 'on' command it will cause AC to switch off, if already on. The problem here is, that this whole integration only assume states of the device, and therefore to be sure 'on' and 'off' commands are send always. 

I could add condition, than when on and off commands are same, than if will not be send if device state is already assume i the required state. Would this solve your issue?

**Comment by klimofey** [comment]
Yes, this will partially solve the problem
If you use this approach, then you need to add a point that when you turn on the AC from the off state, you need to send two commands. First the "on" command, and then the command for a specific temperature state and other things. If already enabled, then yes, only the command to switch the temperature state. And to turn off - only shutdown commands

This will be simpler than my approach, but it seems to me that if the AC enable command already contains the necessary characteristics, then calling two commands instead of one is a crutch :)

**Comment by litinoveweedle** [comment]
Hello, would you please care to open feature request for not sending "off" and "on" command if those are same and device is already in the required assumed state? I would like to implement it, but I would like to be requester and to test it if possible. I will close this PR meanwhile.

**Comment by litinoveweedle** [comment]
Perfect, thank you so much. I will create dedicated branch and let you know when it will be ready for tests.

**Comment by javierugarte** [comment]
Hello! I have the same problem and I'm surprised that there aren't more people experiencing it. Usually, the on/off button is the same.

If you want, I can also support you with any necessary tests.

Best regards,
